File: NBlast.Api/SchedulerConfig.fs
namespace NBlast.Api

open NBlast.Storage.Core
open NBlast.Api.Async
open FluentScheduler
open FluentScheduler.Model
open Microsoft.Practices.Unity
open System

    type TaskFactory(container: IUnityContainer) =
        interface ITaskFactory with
            member me.GetTaskInstance() =
                container.Resolve<ITask>()

    type ScheduleRegistry(?minutes: int) as me =
        inherit Registry()
        do
            let minutes = defaultArg minutes 1 
            me.Schedule<QueueProcessingTask>().ToRunNow().AndEvery(minutes).Minutes() |> ignore
            
    module SchedulerConfig = 
        let private logger = NLog.LogManager.GetCurrentClassLogger()

        let private onTaskException s (e: UnhandledExceptionEventArgs) =
            logger.Fatal(e.ExceptionObject) |> ignore

        let Configure(container: IUnityContainer) =
            logger.Debug("Start to configure task scheduler")

            let configReader = container.Resolve<IConfigReader>()
            let minutes      = configReader.ReadAsInt("NBlast.indexing.scheduler.run_every_minutes")
            let registry     = new ScheduleRegistry(minutes)
            let eventHandler = new GenericEventHandler<TaskExceptionInformation, UnhandledExceptionEventArgs>(onTaskException)

            logger.Debug("Schedule task runner for every {0} minute(s)", minutes)

            TaskManager.TaskFactory <- new TaskFactory(container)
            TaskManager.add_UnobservedTaskException(eventHandler)
            TaskManager.Initialize(registry)

File: NBlast.Rest/Configuration/SchedulerConfig.cs
using System;
using FluentScheduler;
using FluentScheduler.Model;
using LanguageExt;
using Ninject;
using Serilog;
using static LanguageExt.Prelude;

namespace NBlast.Rest.Configuration
{
    public class SchedulerConfig
    {
        private static readonly ILogger Logger = Log.Logger.ForContext<SchedulerConfig>();
        private static readonly string RunEveryMinutes = "NBlast.indexing.scheduler.run_every_minutes";

        private static void OnTaskException(TaskExceptionInformation sender, UnhandledExceptionEventArgs args)
        {
            var ex = args.ExceptionObject as Exception;
            if (ex == null) return;
            Logger.Fatal(ex, ex.Message);
        }

        public static Unit Configure(IKernel kernel)
        {
            Logger.Debug("Start to configure task scheduler");

            var configReader = kernel.Get<IConfigReader>();
            var minutes      = configReader.ReadAsInt(RunEveryMinutes);
            var registry     = new SchedulerRegistry();
            
            Logger.Debug($"Schedule task runner for every {minutes} minute(s)");

            TaskManager.TaskFactory = new TaskFactory(kernel);
            TaskManager.UnobservedTaskException += OnTaskException;
            TaskManager.Initialize(registry);
            return unit;
        }
    }
}